### PR TITLE
[dns-sd] fix SRP service name length

### PR DESCRIPTION
**Pull request by wgtdkp**
#### Problem
 The SRP service name buffers are too small.

 #### Summary of Changes
 Fix SRP service & host name buffer issue.

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

File: src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h [hunk 1]
```diff
@@ -131,14 +131,14 @@ class GenericThreadStackManagerImpl_OpenThread
         struct Service
         {
             otSrpClientService mService;
-            char mInstanceName[kMaxInstanceNameSize];
-            char mName[kMaxNameSize];
+            char mInstanceName[kMaxInstanceNameSize + 1];
+            char mName[kMaxNameSize + 1];
```
Comment (jmartinez-silabs) [inline src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h:135]: Here i think mName already has the NULL implicitly in kMaxNameSize
`static constexpr uint8_t kMaxNameSize         = chip::Mdns::kMdnsTypeMaxSize + chip::Mdns::kMdnsProtocolTextMaxSize + 1;`

Just below kMaxHostNameSize is a magic number 32 ? 
`static constexpr uint8_t kMaxHostNameSize     = 32;` Are we sure its needsa +1 for the Null character or is int already included in the + 32. I think the only verify that was falling is the one testing mInstanceName as that one didn't include the Null char
`static constexpr uint8_t kMaxInstanceNameSize = chip::Mdns::kMdnsNameMaxSize;`

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by andy31415** [comment]
On this PR I see the 'run unit tests' stuck in the run state (without output unfortunately) for 50 minutes. Unsure if PR-specific or a flake. Given that every one of the runners does this, it may be something in the change itself.

**Comment by andy31415** [comment]
I cancelled the workflows with the intent of restarting them, however I do not seem to get a 'restart all' button yet.
If I am unable to do so, @wgtdkp could you push a minor update to kick the presubmits again?

**Review by mspang — APPROVED** [review]

**Comment by woody-apple** [comment]
I kicked the Tests again, for some reason they're getting stuck...

**Comment by andy31415** [comment]
> I kicked the Tests again, for some reason they're getting stuck...

I am totally puzzled here - I have never seen these tests hang on any other PR, however I am unclear on why this PR would cause system tests to misbehave either (and across all compilers to boot). I did copy the PR and tried locally, it passes (except I found a Mdns hang which I fixed in #6721). Kangping reported his workstation hangs on TestSystemObject but I could not see that.

**Comment by wgtdkp** [comment]
@woody-apple @andy31415 Thanks for helping resolve the test hang issue. I am sorry that it is too later for me to fix this issue yesternight, but it seems the hang issue is caused by the failure of the `TestMdns` test case and it is my change in this PR that is failing the `TestMdns` case :(

I submitted #6771 to fix the issue.

**Pull request merged**